Rebase/move work from tswfi #1236 - code coverage in unit tests

thomasnares opened the pull request:
<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop developer documentation! 

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines on how to contribute:
https://devdocs.prestashop-project.org/8/contribute/documentation/how/
------------------------------------------------------------------------------>

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 8.x
| Description?  | Re-integrate work from @tswfi from a previous pending PR #1236 

MERGE AFTER #1661 since we link to the new Docker guide

<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! -->

Commit:
Use PrestaShop with Docker

Commit:
Installation guide with Docker

Commit:
Apply suggestions from review, and change docker-compose references to docker compose

thomasnares committed:
Update modules/testing/ci-cd.md

Commit:
Rebase/move work from tswfi #1236

Commit:
Rebase/move work from tswfi #1236

StoreCommanderDev committed:
Update apache version

It would be great to inform developpers to use the latest version of apache instead of an obsolete version.
Because latest release of Apache 2.2 is 2017 and migrating from 2.2 to 2.4 cause some trouble.

kpodemski committed:
Update basics/installation/system-requirements.md

Co-authored-by: Thomas NARES <thomasnares@gmail.com>

kpodemski committed:
Merge pull request #1670 from StoreCommanderDev/system-requirements-apache-version

Update apache version

Commit:
Add note about date_add and date_upd

Commit:
Add note about authorisation - CGIPassAuth

Commit:
Fix property declaration

matthieu-rolland commented on testing/unit-tests/how-to-execute-tests.md:
````diff
@@ -35,3 +35,40 @@ vendor/bin/phpunit --debug -c tests/Unit/phpunit.xml tests/Unit/PrestaShopBundle
+```
+
+{{% notice tip %}}
+You may use a [Dockerized environment to run the project]({{< relref "/8/basics/installation/install-with-docker" >}}).
````
Apparently the test is failing because this reference is not found

thomasnares replied:
thanks, i left a comment (in the PR desc) about the necessity to merge #1661 before this one since i reference a pending PR 👍

matthieu-rolland reviewed (commented):
small feedback about failing test

matks committed:
Redirect security policy

matks committed:
Merge pull request #1671 from matks/redirect-to-site

Redirect security policy

Commit:
Add notice about formatingh

Commit:
Improve WS docs, internal linking, ...

thomasnares committed:
Update webservice/tutorials/testing-webservice-postman.md

thomasnares committed:
Apply suggestions from code review

Co-authored-by: Krystian Podemski <kpodemski@users.noreply.github.com>

kpodemski committed:
Merge pull request #1673 from thomasnares/demoWs

Improve Webservices section of the documentation

Commit:
Update basics

Commit:
Merge branch 'dockerInstallation' of github.com:thomasnares/ps-docs into dockerInstallation

Samar-al committed:
Update tutorial.md

I thing there's been a confusion between camelCase and PascalCase. CamelCase is a convention used in programming and computer-related fields to write compound words or phrases where each word within the compound starts with a capital letter, except for the first word. In PascalCase, also known as UpperCamelCase, each word within a compound starts with an uppercase letter, including the first word. "MyModule" is in PascalCase, in your doc, it's defined as camelCase.

matthieu-rolland committed:
Merge pull request #1674 from Samar-al/patch-1

Update tutorial.md

prettytree committed:
Fix how GitHub is spelled

prettytree committed:
Fix how GitHub is spelled

nicosomb committed:
Merge pull request #1675 from prettytree/prettytree-patch-1

Fix how GitHub is spelled

JoseNOM committed:
Spell correctly GitHub

matks committed:
Merge pull request #1677 from JoseNOM/patch-1

Spell correctly GitHub

thomasnares committed:
Apply suggestions from code review

Co-authored-by: Krystian Podemski <kpodemski@users.noreply.github.com>

mflasquin committed:
Update documentation

Commit:
Add some details, apply suggestions

Commit:
Add second info about fpm

kpodemski committed:
Merge pull request #1661 from thomasnares/dockerInstallation

PrestaShop on Docker guide

Commit:
Rework installation guide

Commit:
Update system requirements, add NVM mention

Commit:
typos

Commit:
CLI menu title

kpodemski committed:
Merge pull request #1678 from mflasquin/update-doc/8

Update CQRS documentation

matks committed:
Add link to BC Break ADR

kpodemski committed:
Merge pull request #1680 from matks/add-adr-bc-break

thomasnares committed:
Apply suggestions from code review

Co-authored-by: Krystian Podemski <kpodemski@users.noreply.github.com>

thomasnares committed:
Update basics/installation/environments/unix-specific.md

kpodemski committed:
Merge pull request #1679 from thomasnares/improvedInstallationGuide

Improved installation guide

kpodemski commented:
I guess you can rebase it, so we can merge?

nenes25 committed:
Add documentation for hook actionListModules

kpodemski committed:
Apply suggestions from code review

kpodemski committed:
Merge pull request #1681 from nenes25/actionListModules

Add documentation for hook actionListModules

Commit:
Merge branch 'codeCoverage' of github.com:thomasnares/ps-docs into codeCoverage
